### PR TITLE
implement LDAP authentication

### PR DESCRIPTION
**Pull request by intamyuto**
Implemented LDAP authentication through SASL transport. Added cgo dependency with freddierice/go-sasl. 

To build this project you must have libsasl2 installed.

On Debian/Ubuntu: 
```bash
sudo apt-get install libsasl2-dev
```

On Redhat:
```bash
sudo yum install cyrus-sasl-devel.x86_64
```

### CODE REVIEW

**Comment by vishjosh** [comment]
Would love to see a windows build too ! @intamyuto

**Comment by intamyuto** [comment]
Removed cgo dependency. Should cross-compile now.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+
+    con, err := impalathing.Connect(host, port, opts)
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    query, err := con.Query("<sql-query>")
+
+    startTime := time.Now()
+    results := query.query.FetchAll()
```
Comment (paribhasha) [inline README.md]: typo : extraneous query, should be query.FetchAll()

File: README.md [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+    port := 21000
+
+    opts := impalathing.DefaultOptions
+
+    // enable LDAP authentication:
+    opts.UseLDAP = true
+    opts.Username = "<username>"
+    opts.Password = "<password>"
+
+    con, err := impalathing.Connect(host, port, opts)
```
Comment (paribhasha) [inline README.md]: should be con, err := impalathing.Connect(host, port, &opts)

**Comment by intamyuto** [comment]
Fixed README listing errors

File: README.md (hunk 1 above)
Comment (intamyuto) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (intamyuto) [reply README.md]: fixed

**Pull request merged**